vowpalwabbit/learner.h

```diff
@@ -690,7 +700,32 @@ struct reduction_learner_builder
     return *this;
   }
 
-  learner<DataT, ExampleT>* build() { return this->_learner; }
+  learner<DataT, ExampleT>* build(VW::io::logger* logger = nullptr)
+  {
+    if (logger != nullptr)
```

Comment:
Why bother making this optional? Just as a step to ease migration?

Reply:
Yes I was planning to take it out once types are added and the logger is input for each build